pkg/authorization/api/types.go

```diff
@@ -0,0 +1,110 @@
+package api
+
+import (
+	kapi "github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	kruntime "github.com/GoogleCloudPlatform/kubernetes/pkg/runtime"
+	kutil "github.com/GoogleCloudPlatform/kubernetes/pkg/util"
+)
+
+// shouldn't store the information on the users or teams directly because assigning policy is distinct from creating users (or is it?)
+// PolicyRules can be created from a PolicyRuleTemplate plus user/team/account.  PolicyRules can be created from PolicyRuleTemplates
+// in the same names or the global namespace, provided you have rights to read said policies
+// Roles are sets of PolicyRuleTemplates plus the information required to actualize the PolicyRuleTemplates into PolicyRules
+// Authorization is calculated against
+//   1. all allow PolicyRules in the namespace
+//   2. all allow actualized Role PolicyRules in the namespace
+//   3. all allow PolicyRules in the global namespace
+//   4. all allow actualized Role PolicyRules in the global namespace
+//   5. all deny PolicyRules in the namespace
+//   6. all deny actualized Role PolicyRules in the namespace
+//   7. all deny PolicyRules in the global namespace
+//   8. all deny actualized Role PolicyRules in the global namespace
+//   Any allow success short circuits to the denies.  Any deny returns a deny.  Return errors include both the allowing rule and the denying rule.
```

Comment:
I think this description is out of date... I don't see PolicyRuleTemplate objects anywhere, and I think we settled on a different evaluation order (for efficiency, and to make sure global allows don't get trumped by local denies)